### ISSUE TITLE
Using a yml config file causes spurious lines to be added

### ISSUE THREAD

**Issue by AndyBan** [dotansimha/graphql-code-generator#1281]
Using a codegen.yml config file causes spurious lines to be written to the generated file.  

codegen.yml :

```
schema:
  - 'schema.json'
documents:
  - './src/**/*.gql'
  - './src/**/*.graphql.ts'
overwrite: true
config:
  printTime: true
  noNamespaces: true
  resolvers: false
  noGraphqlTag: true
  scalars:
    Milliseconds: number
generates:
  ./src/typings/GraphAPI.ts:
    plugins:
      - add:
          - '// THIS IS A GENERATED FILE, DO NOT EDIT IT!'
          - '// tslint:disable'
      - 'time'
      - 'typescript-common'
      - 'typescript-client'
      - 'typescript-server'

require: []
```

At the top of the generated file there is the extra lines:

```
 
// THIS IS A GENERATED FILE, DO NOT EDIT IT!
// tslint:disable
true
true
true
[object Object]
// Generated in 2019-02-06T10:03:25+00:00
export type Maybe<T> = T | null;
// Snip generated correct code
 
```

The trues have been there some time and could be ignored but this isn't the case for the object tostring which has to be manually removed.

**Environment:**
 - OS: Windows
 - Codegen: 0.16.1
 - Node: 0.11.9

**Comment by dotansimha** [comment]
Hi @AndyBan ! Thank you for reporting it. 
Can you please try to remove the entire `config` section and share the output? I think it might be related to some configuration we changed, but I'm not sure.

**Comment by AndyBan** [comment]
Hi @dotansimha 

If I remove the config section then it generates correctly:

`// THIS IS A GENERATED FILE, DO NOT EDIT IT!
// tslint:disable
export type Maybe<T> = T | null;`


If I generate it without the scalars bit then it generates as:
`// THIS IS A GENERATED FILE, DO NOT EDIT IT!
// tslint:disable
true;
true;
true;
export type Maybe<T> = T | null;`

Whilst still a bit wrong typescript does ignore the trues.  Its almost as if something is looping over the config entries and writing them out.

**Comment by selaux** [comment]
This also happens for JSON configuration.

I investigated the issue a little and it seems that the add plugin is expecting only `string` or `string[]` as input, but gets the whole config object, with top level config merged into it. Then it continues to print all config variable values into the generated file.

A workaround currently is to specifify configuration for each plugin individually, e.g.:

```yaml
schema:
  - 'schema.json'
documents:
  - './src/**/*.gql'
generates:
  ./src/typings/GraphAPI.ts:
    plugins:
      add:
          - '// THIS IS A GENERATED FILE, DO NOT EDIT IT!'
          - '// tslint:disable'
      time:
        printTime: true
      typescript-common:
        noNamespaces: true
        scalars:
          Milliseconds: number
      typescript-client:
        noGraphqlTag: true
        scalars:
          Milliseconds: number
      typescript-server:
        resolvers: false
        scalars:
          Milliseconds: number
require: []
```

**Comment by dotansimha** [comment]
Fixed in the recent refactor.

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍
